hardware/OpenCustomizationSelector/src/com/sonymobile/customizationselector/ImsSwitcher.java

```diff
@@ -22,7 +22,7 @@ public ImsSwitcher(Context context) {
     public void switchOnIMS(int subID) {
         CSLog.d(TAG, "switching IMS ON");
         // Need to reset configuration preference in order to allow reboot dialog to appear.
-        if (!context.getSharedPreferences(Configurator.PREF_PKG, Context.MODE_PRIVATE).edit().putString(Configurator.OLD_CONFIG_KEY, "").commit()) {
+        if (!context.createDeviceProtectedStorageContext().getSharedPreferences(Configurator.PREF_PKG, Context.MODE_PRIVATE).edit().putString(Configurator.OLD_CONFIG_KEY, "").commit()) {
```

Comment:
Note that I changed that line already which causes the conflict: https://github.com/Flamefire/android_device_sony_yoshino-common/commit/f32e54d756e2b4a87ef5921f25c8af802d073673

It is now simply `new Configurator(mContext, null).clearConfigurationKey();` such that all handling of that `OLD_CONFIG_KEY` and the protected-or-not storage context is done in one place (as far as possible) to avoid issues like this. Good catch BTW!